### PR TITLE
Upate Lyft CNI to v0.5.1

### PR DESCRIPTION
**Pull request by maruina**
This PR:
- Update the Lyft CNI to v0.5.1
- Add a new IAM policy because `ec2:DescribeVpcs is required for m5 and c5 instances because the AWS metadata server does not return the secondary CIDR block on these instance types.`
- Add the Lyft CNI to the documentation

~Marked as WIP because I still need to test this build by spinning up a new cluster, but I'm happy to get any feedback if you something that doesn't look right.~
Tested by creating a new cluster and upgrading an existing one.

### CODE REVIEW

File: docs/networking.md [hunk 1]
```diff
@@ -473,6 +474,75 @@ Here are some steps items that will confirm a good CNI install:
 The sig-networking and sig-cluster-lifecycle channels on K8s slack are always good starting places
 for Kubernetes specific CNI challenges.
 
+#### Lyft CNI
+
+The [lyft cni-ipvlan-vpc-k8s](https://github.com/lyft/cni-ipvlan-vpc-k8s) plugin uses Amazon Elastic Network Interfaces (ENI) to assing AWS-managed IPs to Pods using the Linux kernel's IPvlan driver in L2 mode.
```
Comment (dbyron0) [inline docs/networking.md]:
```suggestion
The [lyft cni-ipvlan-vpc-k8s](https://github.com/lyft/cni-ipvlan-vpc-k8s) plugin uses Amazon Elastic Network Interfaces (ENI) to assign AWS-managed IPs to Pods using the Linux kernel's IPvlan driver in L2 mode.
```

**Comment by phyllisstein** [comment]
Particularly jazzed about the documentation! I might propose adding a note about the creating a VPC in advance and setting `spec.networkID`. Probably should have been self-evident from the CNI docs, but I wound up struggling to get the sequence straight anyway.

**Comment by maruina** [comment]
This has been tested in the following scenario:
- updating a cluster running lyft 0.4.0 to 0.5.0
- spinning up a lyft 0.5.0 cluster

I think the only thing missing before merging this is where to store the binaries :)

**Comment by maruina** [comment]
@phyllisstein I updated the docs, please tell me if it makes more sense

**Comment by phyllisstein** [comment]
Thanks for indulging my confusion, and thanks again for the extra docs.

**Comment by maruina** [comment]
The binary has been built by Lyft, so we can review (and hopefully merge) this.

**Comment by rifelpet** [comment]
Thanks for sticking with it!

/lgtm

**Pull request merged**